core/templates/dev/head/components/RuleTypeSelectorDirective.js

```diff
@@ -82,7 +82,12 @@ oppia.directive('ruleTypeSelector', [function() {
         $(select2Node).select2('val', $scope.localValue);
 
         $(select2Node).on('change', function(e) {
+          $scope.localValue = e.val;
+          // Update containing scope
+          $scope.$apply();
           $scope.onSelectionChange()(e.val);
+          // Propagate that change down
```

Comment:
Add more detail in this comment:

// This is needed to propagate the change and display input fields for parameterizing the rule. Otherwise, the input fields do not get updated when the rule type is changed.